### ISSUE TITLE
Command "dotnet tool update" fails if root directory contains several projects

### ISSUE THREAD

**Issue by kkostenkov** [muno92/resharper_inspectcode#482]
Command `dotnet tool update` is run [as soon as Action starts](https://github.com/kk-homa/resharper_inspectcode/blob/main/src/main.ts#L9) and due to [a very old bug](https://github.com/dotnet/sdk/issues/7226) in the tool it fails if the directory contains multiple projects.

In my case this is a game changer because some Unity projects have `.sln` and `.csproj` files in the root of the repo.

Maybe the parsing and execution of workingDirectory parameter could be moved to the beginning of the action?  Then maybe that parameter could be used as a workaround with the value of `./..`

**Comment by kkostenkov** [comment]
[This branch](https://github.com/kk-homa/SampleDotNetProject/tree/linter-multi-project-fails-to-run) could be used for the repro

**Comment by muno92** [comment]
@kkostenkov 
Sorry for slow responding.

After version 1.11.13, workingDirectory parameter is executed on the begining of the action.

Is your problem is solved?

**Comment by kkostenkov** [comment]
Thank you so much for the reply. Meanwhile I did the investigation on my own and it appears that the `dotnet tool update` command is present in two places in the repo. (My lack of knowledge did not allow me to understand why)
And the first execution of it comes before the `main.ts` flow and fails even if I change the working directory.   

Yesterday I did another iteration to solve the matter and it appears that [those changes](https://github.com/muno92/resharper_inspectcode/compare/main...kk-homa:resharper_inspectcode:fix/multi-project-setup) help:
1. Moving workingDirectory parameter parsing
2. Adding `await` to `process.chdir(workingDir)`
And the most important 
3. [Adding](https://github.com/muno92/resharper_inspectcode/compare/main...kk-homa:resharper_inspectcode:fix/multi-project-setup#diff-3d2b59189eeedc2d428ddd632e97658fe310f587f7cb63b01f9b98ffc11c0197R48) `process.chdir(`..`)` to Installer class in index.js . This is the other place of calling `dotnet tool update`

I personally have no idea why 1 & 2 are needed since I [managed to make it work without using workingDirectory parameter](https://github.com/kk-homa/SampleDotNetProject/actions/runs/9960110505/job/27518803117) in the end, but without those changes the workflow kept failing.

So to answer your question directly: no. Simply moving the execution of workingDirectory parameter won't help to solve the issue.
